### PR TITLE
fix: tidy up x/foundation

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

This patch addresses the following items:
- Apply lint to the test files
- Fix minor document errors
- Add a test for collecting treasury
- Change error to panic where the error is not caused by user

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/foundation/keeper/grpc_query.go [hunk 1]
```diff
@@ -206,16 +206,16 @@ func (s queryServer) Grants(c context.Context, req *foundation.QueryGrantsReques
 		_, err := query.Paginate(grantStore, req.Pagination, func(key []byte, value []byte) error {
 			var authorization foundation.Authorization
 			if err := s.keeper.cdc.UnmarshalInterface(value, &authorization); err != nil {
-				return err
+				panic(err)
```
Comment (dudong2) [inline x/foundation/keeper/grpc_query.go:209]: Why did you change "return err" with "panic(err)"?
Is it because if you process it with return, you have to process each of them outside the function?

Comment (0Tech) [reply x/foundation/keeper/grpc_query.go:209]: The error is from `UnmarshalInterface`, which means the state is already corrupted. It's a critical bug and cause by the developer (in this case, me), so there is no reason to return it to the users. It also hides the cause of the error from the user, which is good from the security perspective.

File: x/foundation/keeper/grant.go [hunk 2]
```diff
@@ -57,23 +55,21 @@ func (k Keeper) GetAuthorization(ctx sdk.Context, granter string, grantee sdk.Ac
...
 
-func (k Keeper) setAuthorization(ctx sdk.Context, granter string, grantee sdk.AccAddress, authorization foundation.Authorization) error {
+func (k Keeper) setAuthorization(ctx sdk.Context, granter string, grantee sdk.AccAddress, authorization foundation.Authorization) {
 	store := ctx.KVStore(k.storeKey)
 	key := grantKey(grantee, authorization.MsgTypeURL(), granter)
 
 	bz, err := k.cdc.MarshalInterface(authorization)
 	if err != nil {
-		return err
+		panic(err)
```
Comment (zemyblue) [inline x/foundation/keeper/grant.go:70]: Is this `panic` no problem? Because this `setAutorization` function is call in `Grant` function of `foundation` Msg.

Comment (0Tech) [reply x/foundation/keeper/grant.go:70]: No problem. The server won't panic, and it just converts it to the error. You can also see them on many other places (e.g. functions like `MustMarshal()` actually panics inside it).

**Review by zemyblue — APPROVED** [review]
LGTM

**Review by dudong2 — APPROVED** [review]
